benchmark/result/result.py

```diff
@@ -0,0 +1,48 @@
+"""Basic result classes for benchmarking."""
+
+from typing import Any
+from uuid import UUID
+
+from attrs import define, field
+from attrs.validators import instance_of
+from pandas import DataFrame
+
+
+@define(frozen=True)
+class Result:
+    """A single result of the benchmarking."""
+
+    name: str = field(validator=instance_of(str))
```

Comment:
In my opinion, a `Result` should not have a `name` (see my comment in the `Benchmark` class). However, it is very relevant to know to which `Benchmark` a result corresponds to (which is my guess why you introduced this?). I would however prefer to have a field `corresponding_benchmark` (or with a better name) which could then store the information of the corresponding `Benchmark` that created the result. The minimal solution for this would just be a `str` of the `Benchmark` class name, but it could also contain more information. I'll let you figure out something ;)

Reply:
I guess that my request here is already been solved by the UUIDs right? Resp., this should probably be renamed to `benchmark_name` or something similar.

Reply:
But now that I think of it.... is this even necessary? It basically contains the same information as the UUID, namely the benchmark this result belongs to.

Reply:
I added it to the metadata instead since I have the feeling that it is more reasonable. Looking forward to you opinion.

Reply:
Looks reasonable to me. Still need to have a detailed look at the benchmark object, but resolving this comment here.